src/subgradients.jl

```diff
@@ -0,0 +1,34 @@
+
+@doc raw"""
+    subgrad_distance(M, q, p[, atol=eps(eltype(p))])
+    subgrad_distance!(M, X, q, p[, atol=eps(eltype(p))])
+
+compute the subgradient of the distance (in place of `X`)
+
+```math
+f(p) = d_{\mathcal M}(p, q)
+```
+
+to a fixed point `q` on the manifold `M`.
+The subgradient at `p = q`is given by a norm one tangent vector at `p`.
+
+# Optional
+
+* `atol` – (`eps(eltype(p))`) the tolerance to use when evaluating the distance between `p` and `q`.
+"""
+function subgrad_distance(M, q, p; atol = eps(eltype(p)))
+    @assert distance(M, q, p) ≤ atol ["Since p and q are different, use grad_distance."]
+    X = rand(M; vector_at = p)
+    (norm(M, p, X) > 1.0) && (X ./= norm(M, p, X))
+    return X
+end
+function subgrad_distance!(M, X, q, p; atol = eps(eltype(p)))
+    c = check_vector(M, p, X; atol = atol)
```

Comment:
We usually are “lazy in checking” for performance reasons, that is we assume, that `X` is just fine that we get, no need to have this check (if so we would also prefer to error and not print, but this can also just be removed.

And – aren't all vectors ≤ 1 just fine and not only the ones with norm exactly 1? I would prefer a normal-cone approach.